processor/src/parallel/core_trace_fragment/mod.rs

```diff
@@ -720,7 +721,8 @@ impl OperationHelperRegisters for TraceGenerationHelpers {
     fn op_u32split_registers(hi: Felt, lo: Felt) -> [Felt; NUM_USER_OP_HELPERS] {
         let (t1, t0) = split_u32_into_u16(lo.as_int());
         let (t3, t2) = split_u32_into_u16(hi.as_int());
-        let m = (Felt::from(u32::MAX) - hi).inv();
+        let diff = Felt::from(u32::MAX) - hi;
+        let m = if diff == ZERO { ZERO } else { diff.inverse() };
```

Comment:
We may want to be careful here and use the new `or zero` method since the default P3 inversion panics when th input is zero

Reply:
Isn't this isomorphic to what this line is doing? See also https://github.com/0xMiden/miden-vm/pull/2457/changes/8ed284b57f51fdb45dbb345236bdbab600a0b406